### PR TITLE
Remove dbt-postgres and dbt-tests-adapter from dbt-core

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
Remove  dbt-tests-adapter and dbt-postgres from dbt-core _but_ preserve the adapter one tests in the functional directory to make sure we don't lose any test coverage.

resolves #

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [ ] I have run this code in development and it appears to resolve the stated issue  
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [ ] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

**Review by emmyoop — COMMENTED** [review]
Have you tested if the release will work without any changes?

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 git+https://github.com/dbt-labs/dbt-adapters.git@main
+git+https://github.com/dbt-labs/dbt-postgres.git@main
```
[the remark this answers is not in the document]
Comment (colin-rogers-dbt) [reply dev-requirements.txt:2]: indeed it will

**Comment by MichelleArk** [comment]
In theory, the `dbt-postgres` tests shouldn't provide us with _additional_ testing signal on dbt-core changes that we wouldn't be able to pick up in our existing functional tests. But the current state of things is less ideal than that 🙃 

For example, [test_dbt_show](https://github.com/dbt-labs/dbt-core/blob/main/tests/adapter/dbt/tests/adapter/dbt_show/test_dbt_show.py) tests adapter-specific functionality, but I don't see another test  in tests/functional that includes the `--limit` argument.

We can't really know the full extent of this without auditing all the tests but we'll definitely lose test coverage by removing them outright at this point, so I agree with the decision to keep them + move to tests/functional.

My main concern is that these will drift from their source definitions in dbt-adapters/dbt-postgres over time as new tests are defined in those directories. Longer term, we should explore options for running the postgres tests on commit / before merge without having to actually define them in dbt-core.

**Comment by colin-rogers-dbt** [comment]
> Have you tested if the release will work without any changes?

@emmyoop testing here: https://github.com/dbt-labs/dbt-core/actions/runs/7759825156/job/21164841687

**Comment by MichelleArk** [comment]
Added a follow-on tech debt issue for dbt-core to delete the adapter tests entirely, and instead trigger dbt-postgres adapter tests to run: https://github.com/dbt-labs/dbt-core/issues/9513

**Pull request merged**